### PR TITLE
chore: Migrate `css-*.test.js` to `node:test`

### PR DESCRIPTION
**Pull request by xMohamd**
## Changes

- See #9873
- Migrates `packages/astro/test/css-*.test.js`

## Testing

All tests pass.

## Docs

N/A

### CODE REVIEW

**Comment by xMohamd** [comment]
According to actions the error in `static-build-frameworks.nodetest.js`, I tried to figure out what an error was, but I couldn't
@mingjunlu can you help? 

```
✖ failing tests:

test at file:/workspaces/astro/packages/astro/test/static-build-frameworks.nodetest.js:20:2
✖ can build preact
  'test did not finish before its parent and was cancelled'

test at file:/workspaces/astro/packages/astro/test/static-build-frameworks.nodetest.js:25:2
✖ can build react
  'test did not finish before its parent and was cancelled'

test at file:/workspaces/astro/packages/astro/test/static-build-frameworks.nodetest.js:32:5
﹣ can build lit # SKIP
  'test did not finish before its parent and was cancelled'

test at file:/workspaces/astro/packages/astro/test/static-build-frameworks.nodetest.js:37:2
✖ can build nested framework usage
  'test did not finish before its parent and was cancelled'

test at file:/workspaces/astro/packages/astro/test/static-build-frameworks.nodetest.js:6:1
✖ Static build - frameworks (4982.706418ms)
  TypeError [Error]: Cannot read properties of undefined (reading 'getCurrentStack')
      at performWork (/workspaces/astro/node_modules/.pnpm/react-dom@18.2.0_react@18.2.0/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:6567:56)
      at /workspaces/astro/node_modules/.pnpm/react-dom@18.2.0_react@18.2.0/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:6904:12
      at scheduleWork (/workspaces/astro/node_modules/.pnpm/react-dom@18.2.0_react@18.2.0/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:78:3)
      at startWork (/workspaces/astro/node_modules/.pnpm/react-dom@18.2.0_react@18.2.0/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:6903:3)
      at renderToNodeStreamImpl (/workspaces/astro/node_modules/.pnpm/react-dom@18.2.0_react@18.2.0/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:7048:3)
      at Object.renderToStaticNodeStream (/workspaces/astro/node_modules/.pnpm/react-dom@18.2.0_react@18.2.0/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:7061:10)
      at file:///workspaces/astro/packages/astro/test/fixtures/static-build-frameworks/dist/renderers.mjs:204:27
      at new Promise (<anonymous>)
      at renderToStaticNodeStreamAsync (file:///workspaces/astro/packages/astro/test/fixtures/static-build-frameworks/dist/renderers.mjs:203:10)
      at async renderToStaticMarkup$1 (file:///workspaces/astro/packages/astro/test/fixtures/static-build-frameworks/dist/renderers.mjs:167:14)
      at async Object.check$1 (file:///workspaces/astro/packages/astro/test/fixtures/static-build-frameworks/dist/renderers.mjs:98:3)
      at async renderFrameworkComponent (file:///workspaces/astro/packages/astro/test/fixtures/static-build-frameworks/dist/chunks/astro_1EJwy7jw.mjs:1110:15)
      at async renderComponent (file:///workspaces/astro/packages/astro/test/fixtures/static-build-frameworks/dist/chunks/astro_1EJwy7jw.mjs:1338:10)
      at async renderChild (file:///workspaces/astro/packages/astro/test/fixtures/static-build-frameworks/dist/chunks/astro_1EJwy7jw.mjs:868:11)
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by mingjunlu** [comment]
@xMohamd Hmm, that's weird. I'm not sure if it's a false alarm. As you can see, [static-build-frameworks.test.js](https://github.com/withastro/astro/pull/10047/files) didn't do anything special during migration.

However, I did encounter the same error message (`test did not finish before its parent and was cancelled`) while migrating `@astrojs/vercel`. After searching for a while, I found that passing an async callback to `describe` seemed to cause this issue before Node.js 18.19.0 and 20.6.1. (See https://github.com/nodejs/node/issues/48845 and https://github.com/nodejs/node-core-test/issues/49)

Since different Node.js versions are used across Astro's CI, a current workaround might be avoiding passing an async callback to `describe`. Try to move the async part to other hooks (e.g., `it`, `before`) and see if the error message goes away. You can refer to [the changes in this file](https://github.com/withastro/astro/pull/10010/files#diff-b53acd841dd4e629050e0ede75bb2c386dbe2154fe393a92e8b53f6088ef0a35L41). Good luck!

**Comment by xMohamd** [comment]
@mingjunlu Thanks, I tried everything you said but still got the same error

File: packages/astro/test/css-dangling-references.nodetest.js [hunk 1]
```diff
@@ -25,7 +26,12 @@ describe("When Vite's preloadModule polyfill is used", async () => {
 
 				if (cssReferences === null) return;
 
-				expect(filePaths).to.contain.members(
+				// expect(filePaths).to.contain.members(
+				// 	cssReferences,
+				// 	filePath + ' contains a reference to a deleted css asset: ' + cssReferences
+				// );
```
Comment (ematipico) [inline packages/astro/test/css-dangling-references.nodetest.js:32]: to delete

**Review by ematipico — APPROVED** [review]

File: packages/astro/test/css-dangling-references.nodetest.js (hunk 1 above)
Comment (xMohamd) [reply packages/astro/test/css-dangling-references.nodetest.js]: done

**Comment by xMohamd** [comment]
@ematipico notice that I reverted `static-build-frameworks` tests to the old version that used chai

**Pull request closed without merging**